pkg/fakerp/fakerp.go

```diff
@@ -24,7 +27,76 @@ import (
 	"github.com/openshift/openshift-azure/pkg/util/vault"
 )
 
-func GetDeployer(log *logrus.Entry, cs *api.OpenShiftManagedCluster) api.DeployFn {
+func debugDeployerError(ctx context.Context, log *logrus.Entry, cs *api.OpenShiftManagedCluster, err error, testConfig api.TestConfig) error {
+	authorizer, err := azureclient.GetAuthorizerFromContext(ctx, api.ContextKeyClientAuthorizer)
+	if err != nil {
+		return err
+	}
+
+	deploymentOperations := azureclient.NewDeploymentOperationsClient(ctx, cs.Properties.AzProfile.SubscriptionID, authorizer)
+
+	operations, err := deploymentOperations.List(ctx, cs.Properties.AzProfile.ResourceGroup, "azuredeploy", nil)
+	if err != nil {
+		log.Warnf("failed to get deployment operations: %v", err)
+		return err
+	}
+
+	for _, op := range operations {
+		if *op.Properties.ProvisioningState == "Succeeded" {
+			continue
+		}
+
+		b, _ := json.MarshalIndent(op, "", "  ")
+		log.Debug(string(b))
+
+		if testConfig.ArtifactDir != "" &&
```

Comment:
if nothing is to be done when ArtifactDir is blank, bail out early from this func instead of querying operations and doing an empty loop run

Reply:
the debug line above is made on a per-operation basis regardless of the setting of ArtifactDir